HLSPlugin/src/com/kaltura/hls/m2ts/PESProcessor.as

```diff
@@ -144,8 +159,64 @@ package com.kaltura.hls.m2ts
             
             return true;
         }
-
-        public function append(packet:PESPacket):Boolean
+		
+		public function indexOf(bytes:ByteArray, search:String, startOffset:uint = 0):Number
+		{
+			if (bytes == null || bytes.length == 0) {
+				throw new ArgumentError("bytes parameter should not be null or empty");
+			}
+			
+			if (search == null || search.length == 0) {
+				throw new ArgumentError("search parameter should not be null or empty");
+			}
+			
+			// Fast return is the search pattern length is shorter than the bytes one
+			if (bytes.length < startOffset + search.length) {
+				return -1;
+			}
+			
+			// Create the pattern
+			var pattern:ByteArray = new ByteArray();
```

Comment:
Given this is always ID3, would it be better to just make this loop search for I, D, 3 like we do for some other patterns in the MPEG2 TS stream? Or to at least cache the ByteArray so we don't frequently reallocate it?